pkg/metrics/metrics.go

```diff
@@ -1,143 +1,328 @@
+	resourceKind string, resourceNamespace string, resourceRequestOperation ResourceRequestOperation, ruleName string, ruleResult RuleResult, ruleType RuleType,
+	ruleExecutionCause RuleExecutionCause, log logr.Logger) {
+	ctx := context.Background()
+
+	commonLabels := []attribute.KeyValue{
+		attribute.String("policy_validation_mode", string(policyValidationMode)),
+		attribute.String("policy_type", string(policyType)),
+		attribute.String("policy_background_mode", string(policyBackgroundMode)),
+		attribute.String("policy_namespace", policyNamespace),
+		attribute.String("policy_name", policyName),
+		attribute.String("resource_kind", resourceKind),
+		attribute.String("resource_namespace", resourceNamespace),
+		attribute.String("resource_request_operation", string(resourceRequestOperation)),
+		attribute.String("rule_name", ruleName),
+		attribute.String("rule_result", string(ruleResult)),
+		attribute.String("rule_type", string(ruleType)),
+		attribute.String("rule_execution_cause", string(ruleExecutionCause)),
+	}
+
+	m.policyResultsMetric.Add(ctx, 1, commonLabels...)
+}
+
+func (m *MetricsConfig) RecordPolicyChanges(policyValidationMode PolicyValidationMode, policyType PolicyType, policyBackgroundMode PolicyBackgroundMode, policyNamespace string, policyName string, policyChangeType string, log logr.Logger) {
+	ctx := context.Background()
+
+	commonLabels := []attribute.KeyValue{
+		attribute.String("policy_validation_mode", string(policyValidationMode)),
+		attribute.String("policy_type", string(policyType)),
+		attribute.String("policy_background_mode", string(policyBackgroundMode)),
+		attribute.String("policy_namespace", policyNamespace),
+		attribute.String("policy_name", policyName),
+		attribute.String("policy_change_type", policyChangeType),
+	}
+
+	m.policyChangesMetric.Add(ctx, 1, commonLabels...)
+}
+
+func (m *MetricsConfig) RecordPolicyRuleInfo(policyValidationMode PolicyValidationMode, policyType PolicyType, policyBackgroundMode PolicyBackgroundMode, policyNamespace string, policyName string,
+	ruleName string, ruleType RuleType, status string, metricValue float64, log logr.Logger) {
+
+	ctx := context.Background()
+	commonLabels := []attribute.KeyValue{
+		attribute.String("policy_validation_mode", string(policyValidationMode)),
+		attribute.String("policy_type", string(policyType)),
+		attribute.String("policy_background_mode", string(policyBackgroundMode)),
+		attribute.String("policy_namespace", policyNamespace),
+		attribute.String("policy_name", policyName),
+		attribute.String("rule_name", ruleName),
+		attribute.String("rule_type", string(ruleType)),
+		attribute.String("status_ready", status),
+	}
+
+	m.policyRuleInfoMetric.Observe(ctx, metricValue, commonLabels...)
+}
+
+func (m MetricsConfig) RecordAdmissionRequests(resourceKind string, resourceNamespace string, resourceRequestOperation ResourceRequestOperation, log logr.Logger) {
```

Comment:
Remove logger.

Reply:
This will be removed. Marking as resolved.